docs/spec/v0.5/appl_data_structure.md

```diff
@@ -80,6 +80,20 @@ The data types of function parameters for executable items cannot be determined,
 | b      | binary data (base-64 encoded)                 |
 | u      | utf8-string                                   |
 
+#### Records
+
+It is not always feasible to statically assign IDs for data items at compile-time.
```

Comment:
Maybe it would be worth adding some examples here.
Some that spring to mind are:
* Where the data follows a pattern but the size of the pattern is not known in advance, e.g. a modular system of N modules, or a multi-channel sensor/ADC input where each channel has the same configuration structure.
* If a device is trying to expose data from one or more connected devices (i.e. a protocol bridge)